.github/workflows/ci.yml

```diff
@@ -3,36 +3,32 @@ on: [push, pull_request]
 env:
   VCPKG_COMMIT_HASH: 68d349964cb4e8da561fd849d9491e6ba11c5681
 jobs:
-  createrelease:
-    name: createrelease
+  get_version:
+    name: Get version
     runs-on: ubuntu-latest
-    if: startsWith(github.ref, 'refs/tags/v')
+    outputs:
+      version: ${{ steps.get_version.outputs.version }}
     steps:
       - name: Check out code
         uses: actions/checkout@v4
       - name: Get the version
         id: get_version
-        run: echo "version=${GITHUB_REF/refs\/tags\/v/}" >> $GITHUB_ENV
-      - name: Create release
-        id: create_release
-        uses: softprops/action-gh-release@v2
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          tag_name: v${{ env.version }}
-          name: Release v${{ env.version }}
-          prerelease: false
-      - name: Output Release URL File
-        run: echo "${{ steps.create_release.outputs.upload_url }}" > release_url.txt
-      - name: Save Release URL File for publish
-        uses: actions/upload-artifact@v4
-        with:
-          name: release_url
-          path: release_url.txt
+        run: |
+          # check if git ref matches a tag
+          if [[ "$GITHUB_REF" =~ "refs/tags/v" ]]; then
+            TAG=${GITHUB_REF/refs\/tags\/v/}
+            echo setting output to tag $TAG
+            echo "version=$TAG" >> "$GITHUB_OUTPUT"
```

Comment:
Why change it to `GITHUB_OUTPUT`? If I understand correctly you could have kept writing it to `$GITHUB_ENV` and then read it like: `$version` instead of having to echo it on line 27 and do `needs.get_version.outputs.version`

(I'm basing this comment out of [this link](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#example-of-writing-an-environment-variable-to-github_env)

Reply:
Env is per job, meaning each `[windows, macos, macos-package, release]` would need to have its own tag parsing code and I don't like that.

Reply:
Ah, ok